lib/SPIRV/SPIRVLowerUmulWithOverflow.cpp

```diff
@@ -0,0 +1,158 @@
+//===- SPIRVLowerUmulWithOverflow.cpp - Lower llvm.umul.with.overflow -----===//
+//
+//                     The LLVM/SPIRV Translator
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+// Copyright (c) 2020 Intel Corporation. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal with the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimers.
+// Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimers in the documentation
+// and/or other materials provided with the distribution.
+// Neither the names of Intel Corporation, nor the names of its
+// contributors may be used to endorse or promote products derived from this
+// Software without specific prior written permission.
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// CONTRIBUTORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS WITH
+// THE SOFTWARE.
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements lowering of llvm.umul.with.overflow.* into basic LLVM
+// operations. Probably, in the future this pass can be generalized for other
+// function calls
+//
+//===----------------------------------------------------------------------===//
+#define DEBUG_TYPE "spv-lower-llvm_umul_with_overflow"
+
+#include "LLVMSPIRVLib.h"
+#include "LLVMUmulWithOverflow.h"
+#include "SPIRVError.h"
+#include "llvm/IR/InstVisitor.h"
+#include "llvm/IR/IntrinsicInst.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Verifier.h"
+#include "llvm/IRReader/IRReader.h"
+#include "llvm/Linker/Linker.h"
+#include "llvm/Pass.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Debug.h"
+
+using namespace llvm;
+using namespace SPIRV;
+
+namespace SPIRV {
+cl::opt<bool> SPIRVLowerUmulWithOverflowValidate(
+    "spv-lower-umulwithoverflow-validate",
+    cl::desc("Validate module after lowering llvm.umul.with.overflow.*"
+             "intrinsics"));
+
+class SPIRVLowerUmulWithOverflow
```

Comment:
To me, this seems like an overkill to invent the whole new pass to lower one more intrinsic. I would like to hear feedback from @AlexeySotkin  as well, but personally I would suggest to generalize existing `SPIRVLowerSaddWithOverflow` pass so it can lower more than just a single intrinsic

Reply:
I agree with @AlexeySachkov. If we want to add handling for another instrinsic, we better to generalize `SPIRVLowerSaddWithOverflow` pass,